src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.*;
+import core.basesyntax.enums.Type;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX = 20;
+    private static final int MIN = 1;
+    private static final int DEFAULT_RADIUS = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+
+    public Figure getRandomFigure() {
+        int indexOfType = random.nextInt(Type.values().length);
+        Type type = Type.values()[indexOfType];
+        String color = colorSupplier.getRandomColor();
+        int sideA = random.nextInt(MIN, MAX);
+        int sideB = random.nextInt(MIN, MAX);
+        int sideC = random.nextInt(MIN, MAX);
+        return switch (type) {
+            case SQUARE -> new Square(sideA, color);
+            case RECTANGLE -> new Rectangle(sideA, sideB, color);
+            case TRIANGLE -> new RightTriangle(sideA, sideB, color);
+            case CIRCLE -> new Circle(sideA, color);
+            default -> new IsoscelesTrapezoid(sideA, sideB, sideC, color);
```

Comment:
Instead of returning `null` or relying on a default case that might not be clear, ensure that all possible cases are covered explicitly in your switch statement. If there are no other types, you don't need a default case at all.